packages/esm-patient-chart-app/src/visit/visits-widget/visit.resource.tsx

```diff
@@ -245,3 +276,80 @@ export interface Diagnosis {
     };
   };
 }
+
+export function useDeleteVisit(patientUuid: string, visit: Visit, onVisitDelete = () => {}, onVisitRestore = () => {}) {
+  const { mutateVisits } = useVisits(patientUuid);
+  const { mutate: mutateCurrentVisit } = useVisit(patientUuid);
+  const [isDeletingVisit, setIsDeletingVisit] = useState(false);
+  const { t } = useTranslation();
+
+  const restoreDeletedVisit = () => {
+    restoreVisit(visit?.uuid)
+      .then(() => {
+        showNotification({
+          title: t('visitRestored', 'Visit restored'),
+          description: t('visitRestoredSuccessfully', '{{visit}} restored successfully', {
+            visit: visit?.visitType?.display ?? t('visit', 'Visit'),
+          }),
+          kind: 'success',
+        });
+        mutateVisits();
+        mutateCurrentVisit();
+        onVisitRestore?.();
+      })
+      .catch(() => {
+        showNotification({
+          title: t('visitNotRestored', "Visit couldn't be restored"),
+          description: t('errorWhenRestoringVisit', 'Error occured when restoring {{visit}}', {
+            visit: visit?.visitType?.display ?? t('visit', 'Visit'),
+          }),
+          kind: 'error',
+        });
+      });
+  };
+
+  const initiateDeletingVisit = () => {
+    setIsDeletingVisit(true);
+    const isCurrentVisitDeleted = !visit?.stopDatetime;
+
+    deleteVisit(visit?.uuid)
+      .then(() => {
+        mutateVisits();
+        mutateCurrentVisit();
+        // TODO: Needs to be replaced with Actionable Snackbar when Actionable
+        showActionableNotification({
+          title: !isCurrentVisitDeleted
+            ? t('visitDeleted', '{{visit}} deleted', {
+                visit: visit?.visitType?.display ?? t('visit', 'Visit'),
+              })
+            : t('cancelVisit', 'Cancel visit'),
+          kind: 'success',
+          subtitle: !isCurrentVisitDeleted
+            ? t('visitDeletedSuccessfully', '{{visit}} is deleted successfully', {
+                visit: visit?.visitType?.display ?? t('visit', 'Visit'),
+              })
+            : t('visitCanceled', 'Canceled active visit successfully'),
+          actionButtonLabel: t('undo', 'Undo'),
+          onActionButtonClick: restoreDeletedVisit,
```

Comment:
> Is this used?

Hi @ibacher , yes, so after the visit is `voided`, the notification with an action button `Undo` appears and if the user clicks the `Undo` button, the `restoreDeletedVisit` is called.